app/main/controller/presto_controller.py

```diff
@@ -0,0 +1,23 @@
+from flask import request, current_app as app
+from flask_restplus import Resource, Namespace
+
+from app.main.lib.fields import JsonObject
+from app.main.lib import similarity
+
+api = Namespace('presto', description='presto callback url')
+presto = api.model('presto', {
+    'body': JsonObject(required=False, description='Original body of the message sent - will contain ID, callback_url, and original url/text of request'),
+    'response': JsonObject(required=False, description='The response data sent back from the model'),
+})
+@api.route('/receive/<string:action>/<string:model_type>')
+class PrestoResource(Resource):
+    @api.response(200, 'Successfully responded to presto callback.')
+    @api.doc('Receive a presto callback for a given `model_type`')
+    def post(self, action, model_type):
+        data = request.args or request.json
+        if action == "add_item":
+            return similarity.callback_add_item(data, model_type)
+        elif action == "search_item":
+            return similarity.callback_get_similar_items(data, model_type)
```

Comment:
This is still replying in the "wrong" place as far as I can see @DGaffney . I believe this function is called by Presto returning the audio hash. If Alegre replies here, isn't it replying to Presto (rather than replying to the original caller, e.g., Check API)?

Reply:
Marking as outdated now but if you're still seeing an issue here let me know! I think I've resolved the problem you had raised...